kin-ecosystem-sdk/src/main/java/com/kin/ecosystem/data/blockchain/BlockchainSourceImpl.java

```diff
@@ -156,48 +187,78 @@ public void onError(final Exception e) {
                     @Override
                     public void run() {
                         if (e instanceof AccountNotFoundException) {
-                            callback.onResponse(0);
+                            callback.onResponse(balance.getValue());
                         } else {
                             callback.onFailure(e);
                         }
                     }
                 });
-                Log.d(TAG, "getCurrentBalance onError: " + e.getMessage());
+                Log.d(TAG, "getBalance onError: " + e.getMessage());
             }
         });
     }
 
-    private void setBalance(int balanceValue) {
-        balance.setValue(balanceValue);
-        local.setBalance(balanceValue);
+    @VisibleForTesting
+    void setBalance(final Balance balanceObj) {
+        BalanceUpdate balanceUpdate = balance.getValue();
+        // if the values are not equals so we need to update,
+        // no need to update for equal values.
+        if (balanceUpdate.getAmount().compareTo(balanceObj.value()) != 0) {
+            Log.d(TAG, "setBalance: Balance changed, should get update");
+            balanceUpdate.setAmount(balanceObj.value());
+            balance.postValue(balanceUpdate);
+            local.setBalance(balanceObj.value().intValue());
```

Comment:
local.setBalance(balanceObj.value().intValue()); can be moved out of the if statement